src/components/calendar-with-shortcuts/CalendarWithShortcuts.tsx

```diff
@@ -0,0 +1,132 @@
+import React, { useCallback, useEffect, useMemo, useState } from 'react';
+import { CalendarProps, CalendarWithShortcutProps } from '../../utils/types';
+import { ShortcutBar } from '../shortuct-bar/ShortcutBar';
+import { getAsNewDate, isValid } from '../../utils/date-utils';
+import { ShortcutButtonModel } from '../shortuct-bar/ShortcutButtonModel';
+import Calendar from '../calendar/Calendar';
+
+const styles = {
+  root: {
+    display: 'inline-flex',
+  },
+};
+
+function CalendarWithShortcutsRef(props: CalendarWithShortcutProps, ref: React.Ref<HTMLDivElement>) {
+  const [toggleDateIndex, setToggleDateIndex] = useState<number>(0);
+  const [selectedDate, setSelectedDate] = useState<Date | undefined>(undefined);
+  const [rangeStart, setRangeStart] = useState<Date | undefined>(undefined);
+  const [rangeEnd, setRangeEnd] = useState<Date | undefined>(undefined);
+  const [multiDates, setMultiDates] = useState<Record<string, Date | undefined> | undefined>(undefined);
+
+  const [viewDate, setViewDate] = useState<Date | undefined>(undefined);
+
+  const updateRangeDates = (val: Date[]) => {
+    setRangeStart(val[0]);
+    setRangeEnd(val[1]);
+  };
+
+  const updateSelectedDates = useCallback(
+    (val) => {
+      props.isMultiSelector && typeof Array.isArray(val) && isValid(val[0])
+        ? setMultiDates(val)
+        : props.isRangeSelector && Array.isArray(val) && isValid(val[0])
+        ? updateRangeDates(val)
+        : isValid(val as Date)
+        ? setSelectedDate(val)
+        : () => 0;
+    },
+    [props.isMultiSelector, props.isRangeSelector],
+  );
+
+  const onCalendarChange = useCallback(
+    (val) => {
+      updateSelectedDates(val);
+      if (props.onChange) {
+        props.onChange(val);
+      }
+    },
+    [props, updateSelectedDates],
+  );
+
+  const updateDateView = useCallback(
+    (date: Date | undefined) => {
+      if (isValid(date)) {
+        setViewDate(getAsNewDate(date));
```

Comment:
To allow for a new reference to be sent and cause re-render even if the `viewDate` prop inside CalendarView never changed.